#641 ManageController tests implemented

Chris-Wooder opened the pull request:
#641

Chris-Wooder commented:
Hi, please could somebody help with these few questions? 

- Should I keep adding to the same PR, or should I open a new one so is easier to review/merge.
- Is there a page/document I can see that describes the key business domain objects and concepts for the project.
- There there a slack group or similar for the project I can join?

Thanks,

Chris.

Chris-Wooder commented:
@tonysurma would you be able to help with questions above?  Thanks.

tonysurma commented:
apologies for delay

Generally smaller PRs are better - easier to review and easier to merge/manage

@stevejgordon - do you recall if we have any docs on the key objects yet?

email me at firstname dot lastname at htbox.org and I will add you to the slack channel for allready

Thanks again

tonysurma commented:
Apologies for delays, @mgmccarthy after your review and any updates I will merge.  Thanks all

mgmccarthy commented on AllReadyApp/Web-App/AllReady.UnitTest/Controllers/ManageControllerTests.cs:
```diff
@@ -780,42 +780,123 @@ public void EnableTwoFactorAuthenticationHasValidateAntiForgeryTokenAttribute()
 
-        [Fact(Skip = "NotImplemented")]
+        [Fact]
         public async Task DisableTwoFactorAuthenticationInvokesSetTwoFactorEnabledAsyncWithCorrectParametersWhenUserIsNotNull()
```
const for UserId

mgmccarthy commented on AllReadyApp/Web-App/AllReady.UnitTest/Controllers/ManageControllerTests.cs:
```diff
@@ -780,42 +780,123 @@ public void EnableTwoFactorAuthenticationHasValidateAntiForgeryTokenAttribute()
 
-        [Fact(Skip = "NotImplemented")]
+        [Fact]
         public async Task DisableTwoFactorAuthenticationInvokesSignInAsyncWhenUserIsNotNull()
```
const for UserId

mgmccarthy commented on AllReadyApp/Web-App/AllReady.UnitTest/Controllers/ManageControllerTests.cs:
```diff
@@ -780,42 +780,123 @@ public void EnableTwoFactorAuthenticationHasValidateAntiForgeryTokenAttribute()
 
-        [Fact(Skip = "NotImplemented")]
+        [Fact]
         public async Task DisableTwoFactorAuthenticationRedirectsToCorrectAction()
```
const for UserId

mgmccarthy commented on AllReadyApp/Web-App/AllReady.UnitTest/Controllers/ManageControllerTests.cs:
```diff
@@ -899,14 +899,28 @@ public void DisableTwoFactorAuthenticationHasValidateAntiForgeryTokenAttribute()
+            var result = controller.ChangePassword();
+
+            //Assert
+            Assert.IsType(typeof(ViewResult), result);
```
replace with Assert.IsType<ViewResult>(result);

Chris-Wooder replied:
Hi, assume you mean use the generic version instead? i.e  Assert.IsType<ChangeEmailViewModel>(result);

mgmccarthy commented on AllReadyApp/Web-App/AllReady.UnitTest/Controllers/ManageControllerTests.cs:
```diff
@@ -930,18 +930,60 @@ public void ChangePasswordGetHasHttpGetAttribute()
-            await TaskCompletedTask;
+        {
+            //Arrange
+            ChangePasswordViewModel validVm = new ChangePasswordViewModel { OldPassword = "oldPassword", NewPassword = "newPassword" };
```
use var

mgmccarthy commented on AllReadyApp/Web-App/AllReady.UnitTest/Controllers/ManageControllerTests.cs:
```diff
@@ -930,18 +930,60 @@ public void ChangePasswordGetHasHttpGetAttribute()
 
-        [Fact(Skip = "NotImplemented")]
+        [Fact]
         public async Task ChangePasswordPostSendsUserByUserIdQueryWithCorrectUserId()
```
const for UserId

mgmccarthy commented on AllReadyApp/Web-App/AllReady.UnitTest/Controllers/ManageControllerTests.cs:
```diff
@@ -930,18 +930,60 @@ public void ChangePasswordGetHasHttpGetAttribute()
+            mediator.Setup(m => m.SendAsync(It.IsAny<UserByUserIdQuery>())).ReturnsAsync(new ApplicationUser { Id = "userID" });
+
+
+            ManageController controller = new ManageController(userManagerMock.Object, signInManagerMock.Object, mediator.Object);
```
use var

mgmccarthy commented on AllReadyApp/Web-App/AllReady.UnitTest/Controllers/ManageControllerTests.cs:
```diff
@@ -930,18 +930,60 @@ public void ChangePasswordGetHasHttpGetAttribute()
-            //delete this line when starting work on this unit test
-            await TaskCompletedTask;
+            //Arrange
+            ChangePasswordViewModel validVm = new ChangePasswordViewModel { OldPassword = "oldPassword", NewPassword = "newPassword" };
```
var

Chris-Wooder commented:
Hey @mgmccarthy, thanks for the review and comments. It's taken a while for me to get around to addressing your comments, but it's now done and committed.

Unfortunately I think I've messed up when updating my respiratory and pushing commits back across, as all other commits to the master repository seem to have been included in this PR.  I've little experience with git as I've always used TFS professionally.  Please let me know if I need to do anything with the PR wrt this.

Thanks,
C

mgmccarthy commented:
Assert.IsType<ViewResult>(result);

      From: Chris Wooder <notifications@github.com>
 To: HTBox/allReady <allReady@noreply.github.com> 
Cc: Michael McCarthy <mike_mccarthy3@yahoo.com>; Mention <mention@noreply.github.com>
 Sent: Tuesday, June 13, 2017 12:21 PM
 Subject: Re: [HTBox/allReady] #641 ManageController tests implemented (#1975)
   
@Chris-Wooder commented on this pull request.In AllReadyApp/Web-App/AllReady.UnitTest/Controllers/ManageControllerTests.cs:>          public void ChangePasswordGetReturnsAView()
-        {
+        {
+            var controller = new ManageController(null, null, null);
+
+            //Act
+            var result = controller.ChangePassword();
+
+            //Assert
+            Assert.IsType(typeof(ViewResult), result);
Hi, assume you mean use the generic version instead? i.e Assert.IsType(result);—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or mute the thread.

mgmccarthy commented:
@Chris-Wooder, I'm not very good with Git as I'm unfortunately in the same boat as you (stuck with TFS). As far as how to go about making changes to a branch and making sure you get latest before pushing your commit, AllReady uses a concept called rebasing. Steve Gordon (a regular contributor) wrote a really great blog post on how to do a rebase for AllReady: https://www.stevejgordon.co.uk/contributing-to-allready

It's a great read and really helped me get my head around rebasing

That being said, I THINK the way you pushed your changes is still ok. I know you see a bunch of other changes there that you didn't make, but I think they're there b/c of how you went about getting the latest code merged and addressing the code review findings.

perhaps @tonysurma and/or @MisterJames can weight in here. Is what @Chris-Wooder did for this PR correct?

Thanks

mgmccarthy commented:
@Chris-Wooder all the changes look good. The unit tests run. Even thought I'm still not too sure how you ended up pushing all the other commits that we've merged since your original PR for this item, I think the code is looking code.

@tonysurma :shipit:

tonysurma commented:
thanks @mgmccarthy for the reviews and @Chris-Wooder for the work!  I will have @MisterJames and @stevejgordon chime in if there is any issues on merging this with the 'extra' commits.  I know we have seen that before and don't recall the solution/answer.

Thanks again!

MisterJames commented:
I've just squashed in this case so that the changes, though only reflected in one commit, are representative of the work completed on this PR.  Thanks @Chris-Wooder for your work here, great stuff, and thanks @mgmccarthy for the great review/guidance.
